_pages/dev/ssr/server-side-rendering-contextual-logging.md

```diff
@@ -252,3 +252,100 @@ You can also customize the behavior of the `LoggerService` by providing your own
+      "traceContext": {
+        "version": "00",
+        "traceId": "0af7651916cd43dd8448eb211c80319c",
+        "parentId": "b7ad6b7169203331",
+        "traceFlags": "01"
+      }
+    }
+  }
+}
+```
+
+If the traceContext is not appearing in the logs, there are several potential reasons for this:
+- **W3C Trace Context Not Enabled in Dynatrace:** Ensure that W3C Trace Context is enabled in your Dynatrace configuration. Without proper configuration, trace context information may not be collected.
+- **Middleware Filtering:** Check if any middleware or components in your system are filtering out the `traceparent` header. If it's being removed or modified before reaching your application, the trace context information may not be captured.
+- **Invalid `traceparent` Header:** Verify that the traceparent header value in the incoming request is valid and conforms to the W3C Trace Context specification. An invalid header may not be recognized and processed correctly.
+
+Addressing these potential issues is essential to ensure the correct inclusion of the traceContext in your logs. The built-in parser, designed to extract the trace context from the `traceparent header``, satisfies all security requirements specified in the W3C Trace Context standard. It meticulously examines each component of the header to verify its proper format. If the header is found to be invalid, the traceContext property will not be appended to the logs. Additionally, the parser will generate a log message that can be useful for debugging purposes.
+
+The following is an example of such a log:
+
+```json
+{
+  "message": "Traceparent header has invalid length: 18. Expected 55 characters.",
+  "context": {
+    "timestamp": "2023-09-13T12:14:21.377Z",
+    "request": {
+      "url": "/electronics-spa/en/USD/"
+    }
+  }
+}
+```
+
+Depending on the error ocurred during the parser, following error messages may be generated:
+- `Traceparent header has invalid length: ${traceparent.length}. Expected 55 characters.` - the `traceparent` header has invalid length
+- `Traceparent header has invalid format.` - the `traceparent` header has invalid format. To learn more, see [W3C Trace Context - Traceparent Header](https://www.w3.org/TR/trace-context/#traceparent-header)
+
+### Connecting Logs to Distributed Traces
+
+**Note:** These steps have been verified on SAP Commerce Cloud, where OpenSearch Dashboards is the default monitoring tool. Please be aware that the steps described here may vary when using other monitoring tools.
+
+All logs generated by Spartacus can be seen in OpenSearch Dashboards monitoring tool.
+The following is an example of a log presented the monitoring tool:
+
+![Log with Trace Context](assets/images/../../../../../assets/images/ssr/contextual_logging_trace_context.png)
+
+OpenSearch enables the formatting of fields as hyperlinks, and this section outlines the steps for configuring links that direct you to the Dynatrace distributed trace view.
```

Comment:
```suggestion
OpenSearch allows for formatting the fields as hyperlinks, and this section outlines the steps for configuring clickable links that direct you from a certain log entry to a related distributed trace view in Dynatrace. This might be very convenient when troubleshooting.
```